src/appleseed/renderer/kernel/intersection/assemblytree.cpp

```diff
@@ -791,9 +792,17 @@ bool AssemblyLeafVisitor::visit(
                     m_shading_point.m_object_instance_index = object_instance_index_pair.second;
                     m_shading_point.m_primitive_index = 0;
                     m_shading_point.m_primitive_pa = result.m_material_slot;
-                    m_shading_point.m_geometric_normal = object_instance_transform.normal_to_parent(result.m_geometric_normal);
-                    m_shading_point.m_original_shading_normal = object_instance_transform.normal_to_parent(result.m_shading_normal);
+                    m_shading_point.m_geometric_normal =
```

Comment:
That looks good.

Could you add a simple test scene that demonstrates the bug and shows that normals are now properly transformed?